Don't use -march=native when building with clang

Because of a logic error, lame's configure always passes -march=native
when building with clang.

This is a terrible idea in general as it's not often your build machine
and target hardware aligns, and leads to some amusing errors:

	unknown target CPU 'neoverse-n1'
	note: valid target CPU values are: i386, i486, ...

Move the HAVE_CLANG block up into the FULL_OPTIMIZATION case, alongside
HAVE_GCC.  This option is never enabled (as it's basically "use native")
so resolves the build failure.

Upstream-Status: Inappropriate [obsoleted by "Detect and enable optimization
                 and hardening flags across the builds" upstream in lame-4.1]
Signed-off-by: Ross Burton <ross.burton@arm.com>

diff --git i/configure.ac w/configure.ac
index 1111111..2222222 100644
--- i/configure.ac
+++ w/configure.ac
@@ -1021,45 +1021,43 @@ if test "x$HAVE_GCC" = "xyes" -o "x$HAVE_CLANG" = "xyes"; then
 				;;
 			esac
 		fi
+
+		if test "x${HAVE_CLANG}" = "xyes"; then
+			case "${CLANG_VERSION}" in
+			3.[[89]]*|[[45]].*|1[[0-9]].*)
+				OPTIMIZATION="-Ofast"
+				;;
+			*)
+				OPTIMIZATION="-O3"
+				;;
+			esac
+
+			# generic CPU specific options
+			case ${host_cpu} in
+			*486)
+				OPTIMIZATION="${OPTIMIZATION} -march=i486"
+				;;
+			*586)
+				OPTIMIZATION="${OPTIMIZATION} -march=i586 \
+					-mtune=native"
+				;;
+			*686)
+				OPTIMIZATION="${OPTIMIZATION} -march=i686 \
+					-mtune=native"
+				;;
+			esac
+
+			# if someone supplies own CFLAGS, we don't add our own
+			if test "x${ac_save_CFLAGS}" != "x"; then
+				OPTIMIZATION=""
+			fi
+		fi
 		;;
 	*)
 		AC_MSG_ERROR(bad value �${CONFIG_EXPOPT}� for expopt option)
 		;;
 	esac

-
-	if test "x${HAVE_CLANG}" = "xyes"; then
-		case "${CLANG_VERSION}" in
-		3.[[89]]*|[[45]].*|1[[0-9]].*)
-			OPTIMIZATION="-Ofast"
-			;;
-		*)
-			OPTIMIZATION="-O3"
-			;;
-		esac
-
-		# generic CPU specific options
-		case ${host_cpu} in
-		*486)
-			OPTIMIZATION="${OPTIMIZATION} -march=i486"
-			;;
-		*586)
-			OPTIMIZATION="${OPTIMIZATION} -march=i586 \
-				-mtune=native"
-			;;
-		*686)
-			OPTIMIZATION="${OPTIMIZATION} -march=i686 \
-				-mtune=native"
-			;;
-		esac
-
-		# if someone supplies own CFLAGS, we don't add our own
-		if test "x${ac_save_CFLAGS}" != "x"; then
-			OPTIMIZATION=""
-		fi
-	fi
-
-
 	if test "${expopt_msg_result_printed}" = "no" ; then
 		AC_MSG_RESULT(${CONFIG_EXPOPT})
 	fi
